File: 0657-robot-return-to-origin/0657-robot-return-to-origin.java
class Solution {
    public boolean judgeCircle(String moves) 
    {
        int vertical = 0;
        int horizontal = 0;
        char arr[] = moves.toCharArray();
        
        for(char ch : arr)
        {
            if(ch == 'U')
            {
                vertical++;
            }
            else if(ch == 'D')
            {
                vertical--;
            }
            else if(ch == 'R')
            {
                horizontal++;
            }
            else if(ch == 'L')
            {
                horizontal--;
            }
        }
        
        if(vertical == 0 && horizontal == 0)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

File: 0657-robot-return-to-origin/0657-robot-return-to-origin.cpp
class Solution {
public:
    bool judgeCircle(string moves) 
    {
        int Vertical = 0;
        int Horizontal = 0;
        
        for(auto ch : moves)
        {
            switch(ch)
            {
                case 'U':
                    Vertical++;
                    break;
                case 'D':
                    Vertical--;
                    break;
                case 'R':
                    Horizontal++;
                    break;
                case 'L':
                    Horizontal--;
                    break;
            }
        }
        
        if(Vertical == 0 && Horizontal == 0)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
};